tactics/class_tactics.ml

```diff
@@ -447,7 +446,7 @@ let cut_of_hints h =
 
 let catchable = function
   | Refiner.FailError _ -> true
-  | e -> Logic.catchable_exception e
+  | e -> Logic.catchable_exception e [@@ocaml.warning "-3"]
```

Comment:
It is a bit worrying that we have to keep this here.

Reply:
Maybe we don't need it. I don't understand type classes very well and I was not courageous enough to test if it could be removed.

Reply:
Indeed we should explore that in a different PR.